semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala

```diff
@@ -168,8 +169,15 @@ object Main {
       case sym: ExternalSymbol =>
         // NOTE: Object and package external symbols
         // are indistinguishable from each other.
-        val nameEntryType = sym.entry.scalaSig.table(sym.entry.index + 1)._1
-        val hasTermName = nameEntryType == 1
+        val hasTermName = {
+          val idx = sym.entry.index + 1
+          if (sym.entry.scalaSig.hasEntry(idx)) {
+            val nameEntryType = sym.entry.scalaSig.table(idx)._1
+            nameEntryType == 1
+          } else {
+            false
```

Comment:
Do you happen to have a reproduction?

Reply:
scala-library and scalameta jars reproduce it 100s of times, but I don't have a minimal reproduction. `scala.matching` was one case I remember. Once we have these tests + diff tests against metac we'll have plenty of opportunity to extract minimal reproductions. For now I think it's most important that new changes to metacp are tested against a large input to prevent further blatant crashes.